modules/dnn/CMakeLists.txt

```diff
@@ -31,6 +31,11 @@ if(HAVE_TIMVX)
   ocv_target_compile_definitions(${the_module} PRIVATE "HAVE_TIMVX=1")
 endif()
 
+if(HAVE_CANN)
+  # ocv_target_compile_definitions(${the_module} PRIVATE "HAVE_CANN=1")
+  add_definitions(-DHAVE_CANN=1)
+endif()
```

Comment:
@alalek I found if `HAVE_CANN=1` is added by `ocv_target_compile_definitions`, `HAVE_CANN` is undefined in dnn/test, leading to the failure of `HAVE_*` (CANN, TIMVX, Webnn, ...). Do we have a fix for this issue?

Reply:
We should avoid using of "global" `add_definitions()`.
Use `ocv_target_compile_definitions` for test targets separately (under check `if(TARGET ...)`). See the last lines of this file.

Reply:
Done.